### DISCUSSION TITLE
ScriptReference/UIElements.UQueryBuilder_1.ForEach

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#627] [Notes]
# ScriptReference/UIElements.UQueryBuilder_1.ForEach

https://docs.unity3d.com/ScriptReference/UIElements.UQueryBuilder_1.ForEach.html

**Comment by vertxxyz** [comment]
This is a depth-first search.
Also see [`UQueryState<T0>.ForEach`](https://docs.unity3d.com/ScriptReference/UIElements.UQueryState_1.ForEach.html).
